routes/dashboard.rb

```diff
@@ -1,15 +1,34 @@
 class App < Sinatra::Base 
   get '/' do
     @cities = City.all
-    
-    @routes = Route.all.map do |route|
+
+    slim :index
+  end
+
+  post '/' do
+    if params[:departure_id].nil? || params[:departure_id].empty? || params[:arrival_id].nil? || params[:arrival_id].empty?
+      redirect '/'
+    else
+      query = request.params.map{|key, value| "#{key}=#{value}"}.join('&')
+      redirect "/search/#{query}"
+    end
+  end
+
+  get '/search/departure_id=:departure_id&arrival_id=:arrival_id?' do
+    @routes = Route.where({ start_id: params[:departure_id], destination_id: params[:arrival_id]}).map do |route|
       {
-        from: @cities.find { |c| c.id == route.start_id }.title,
-        departure_datetime: route.departure_datetime,
-        to: @cities.find { |c| c.id == route.destination_id }.title,
-        arrival_datetime: route.arrival_datetime,
+          departure_station: City.find_by_id(params[:departure_id]).title,
+          departure_datetime: route.departure_datetime,
+          arrival_station: City.find_by_id(params[:arrival_id]).title,
+          arrival_datetime: route.arrival_datetime
       }
     end
     slim :index
   end
+
+  get '/cities.json' do
+    content_type :json
+    @cities = City.all
+    @cities.map { |c| { label: c.title, value: c.id } }.to_json
```

Comment:
Why do you need instance variable here?
`City.all.map { |c| ..... }.to_json`

Reply:
Also, consider using `jbuilder`
https://gist.github.com/a2ikm/5072882

Reply:
... or `Serializer`
https://www.sitepoint.com/active-model-serializers-rails-and-json-oh-my/